extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -0,0 +1,148 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {getSocialConfig} from './amp-social-share-config';
+
+const DEFAULT_WIDTH = 60;
+const DEFAULT_HEIGHT = 44;
+
+const CONTAINER_TAG = "span";
+const LINK_TAG = "a";
+
+class AmpSocialShare extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  buildCallback() {
+    this.config = this.getElementConfig(this.element);
+    AMP.assert(this.config, 'The <script> config is required for ' +
+      '<amp-social-share> %s');
+
+    this.type = AMP.assert(this.element.getAttribute('type'),
+        'The type attribute is required for <amp-social-share> %s',
+        this.element);
+    this.typeConfig = getSocialConfig(this.type);
+
+    this.container = this.renderShare(this.config);
+
+    this.width = this.element.getAttribute('width') || DEFAULT_WIDTH;
+    this.height = this.element.getAttribute('height') || DEFAULT_HEIGHT;
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.element.appendChild(this.container);
+    return Promise.resolve();
+  }
+
+  renderShare(config) {
+    const urlParams = {};
+
+    for (const param in this.typeConfig["params"]) {
+      const paramConf = this.typeConfig["params"][param];
+      AMP.assert(!paramConf["required"] || param in config,
+        param + " is a required configuration item for " + this.type);
+
+      if (param in config) {
+        const value = config[param];
+        if ("maxlength" in paramConf) {
+          const maxlength = paramConf["maxlength"];
+          AMP.assert(value.length < maxlength,
+            param + " cannot exceed " + maxlength);
+        }
+        urlParams[paramConf["param"]] = value;
+      } else if (paramConf["type"] == "fixed") {
+        urlParams[paramConf["param"]] = paramConf["value"];
+      }
+    }
+    let container = this.getFirstSubElement(this.element, CONTAINER_TAG);
+    let link = this.getFirstSubElement(this.element, LINK_TAG);
+    if (container === null) {
+      container = this.getWin().document.createElement(CONTAINER_TAG);
+      container.setAttribute("class", "-amp-social-share -amp-social-share-" +
+        this.type);
+    }
+    if (link === null) {
+      link = this.getWin().document.createElement(LINK_TAG);
+      link.innerHTML = this.typeConfig["text"];
+    }
+    container.appendChild(link);
+    if (link.getAttribute("href") === null) {
+      link.setAttribute("href", this.buildUrl(this.typeConfig["url"],
+        urlParams));
+    }
+    if (link.getAttribute("target") === null) {
+      link.setAttribute("target", "_blank");
+    }
+
+    return container;
+  }
+
+  buildUrl(base, params) {
+    const paramNames = Object.keys(params);
+    let url = base;
```

Comment:
why not just reuse the `base` param or name `base` `url` instead of having to re-alias? (since i dont see `base` actually used anywhere else in this method)